hot/style-loader.js

```diff
@@ -0,0 +1,43 @@
+/* eslint-env commonjs */
+const loaderUtils = require(`loader-utils`);
+const helpers = require(`./loader-helpers`);
+
+// Used in non-HMR mode, do nothing
+module.exports = source => source;
+
+module.exports.pitch = function(remainingReq) {
+  if (!helpers.isDevServerHot(this.options)) {
+    return;
+  }
+
+  const moduleId = loaderUtils.stringifyRequest(this, `!!${remainingReq}`);
+  const loaderOpts = loaderUtils.getOptions(this) || {};
+  let moduleSource = ``;
+  let isShadowCss = loaderOpts.shadow || /\b(inline|shadow)\b/.test(this.resourceQuery);
+
+  if (isShadowCss) {
```

Comment:
I think you could probably do this more generically/robustly at runtime by having your generated code look at `elem.getConfig('useShadowDom')` and/or `elem.getConfig('css')`.

Reply:
I like that approach